### PR TITLE
Convert merge errors to bad requests

### LINKED ISSUE #1547: DirtyBranch renders as internal server error
Returning ErrDirtyBranch, currently part of Merge and Revert rendered as HTTP status code 500 with an internal server error.
It is a UX that the end-user will have a hard time understanding the root cause.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1547

### CODE REVIEW

File: api/controller.go [hunk 1]
```diff
@@ -959,19 +959,19 @@ func (c *Controller) MergeMergeIntoBranchHandler() refs.MergeIntoBranchHandler {
...
 		case errors.Is(err, catalog.ErrConflictFound) || errors.Is(err, graveler.ErrConflictFound):
 			payload := newMergeResultFromCatalog(res)
 			return refs.NewMergeIntoBranchConflict().WithPayload(payload)
 		case errors.Is(err, catalog.ErrNoDifferenceWasFound) || errors.Is(err, graveler.ErrNoChanges):
-			return refs.NewMergeIntoBranchDefault(http.StatusInternalServerError).WithPayload(responseError("no difference was found"))
+			return refs.NewMergeIntoBranchBadRequest().WithPayload(responseError("no difference was found"))
 		case errors.Is(err, graveler.ErrLockNotAcquired):
 			return refs.NewMergeIntoBranchDefault(http.StatusInternalServerError).WithPayload(responseError("branch is currently locked, try again later"))
+		case errors.Is(err, graveler.ErrDirtyBranch):
+			return refs.NewMergeIntoBranchBadRequest().WithPayload(responseError("can't merge into a dirty branch"))
```
Comment (guy-har) [inline api/controller.go:972]: I think this one should return `409 conflict`

**Review by guy-har — APPROVED** [review]

File: api/controller.go (hunk 1 above)
Comment (itaiad200) [reply api/controller.go:972]: I think it's confusing to overload the term `conflict`. Nothing really "conflicts" here, it's more of a limitation we introduced and might remove later.

Comment (guy-har) [reply api/controller.go:972]: I understand, The thing is that Bad request is used for client error. I don't think that this is a client error. It seems like it fits the documentation of `409 conflict`

Comment (itaiad200) [reply api/controller.go:972]: But it returns a `merge_result` with conflicts. Should we return an empty object in that case?
Or an empty body? 2 response codes with different payloads is error prune IMO.

**Pull request merged**